mlflow/models/resources.py

```diff
@@ -0,0 +1,145 @@
+    DATABRICKS_LLM_ENDPOINT_NAME = "databricks_llm_endpoint_name"
+    DATABRICKS_CHAT_ENDPOINT_NAME = "databricks_chat_endpoint_name"
+
+
+class Providers(Enum):
+    DATABRICKS = "databricks"
+
+
+@dataclass
+class Resource(ABC):
+    """
+    Base class for all resources to define the resources needed to serve a model.
+
+    Args:
+        type (ResourceType): The resource type.
+        target_uri (str): The target URI where these resources are hosted.
+    """
+
+    type: ResourceType
+    target_uri: Providers
+
+    @abstractmethod
+    def to_dict(self):
+        """
+        Convert the resource to a dictionary.
+        Subclasses must implement this method.
+        """
+
+
+@dataclass
+class DatabricksResource(Resource, ABC):
+    """
+    Base class to define all the Databricks resources to serve a model.
+    """
+
+    target_uri: Providers = Providers.DATABRICKS.value
+
+
+@dataclass
+class DatabricksLLMEndpoint(DatabricksResource):
+    """
+    Define Databricks LLM endpoint resource to serve a model.
+
+    Args:
+        endpoint_name (str): The name of all the databricks endpoints used by the model.
+    """
+
+    type: ResourceType = ResourceType.DATABRICKS_LLM_ENDPOINT_NAME
+    endpoint_name: str = None
+
+    def to_dict(self):
+        return {self.type.value: [self.endpoint_name]} if self.endpoint_name else {}
+
+
+@dataclass
+class DatabricksChatEndpoint(DatabricksResource):
+    """
+    Define Databricks LLM endpoint resource to serve a model.
+
+    Args:
+        endpoint_name (str): The name of all the databricks chat endpoints
+        used by the model.
+    """
+
+    type: ResourceType = ResourceType.DATABRICKS_CHAT_ENDPOINT_NAME
+    endpoint_name: str = None
+
+    def to_dict(self):
+        return {self.type.value: [self.endpoint_name]} if self.endpoint_name else {}
+
+
+@dataclass
+class DatabricksEmbeddingEndpoint(DatabricksResource):
+    """
+    Define Databricks embedding endpoint resource to serve a model.
+
+    Args:
+        endpoint_name (str): The name of all the databricks embedding endpoints
+        used by the model.
+    """
+
+    type: ResourceType = ResourceType.DATABRICKS_EMBEDDINGS_ENDPOINT_NAME
+    endpoint_name: str = None
+
+    def to_dict(self):
+        return {self.type.value: [self.endpoint_name]} if self.endpoint_name else {}
+
+
+@dataclass
+class DatabricksVectorSearchEndpoint(DatabricksResource):
+    """
+    Define Databricks vector search endpoint resource to serve a model.
+
+    Args:
+        endpoint_name (str): The name of all the databricks vector search endpoints
+        used by the model.
+    """
+
+    type: ResourceType = ResourceType.DATABRICKS_VECTOR_SEARCH_ENDPOINT_NAME
+    endpoint_name: str = None
+
+    def to_dict(self):
+        return {self.type.value: [self.endpoint_name]} if self.endpoint_name else {}
+
+
+@dataclass
+class DatabricksVectorSearchIndexName(DatabricksResource):
```

Comment:
Why are these separate classes? Doesn't the chain just call the endpoint?

Reply:
@sunishsheth2009 This was not resolved

Reply:
One is a index and one is the endpoint. 
So in the updated PR, we are just going to have 2 classes.
1. DatabricksVectorSearchIndex: Used to create index
2. DatabricksServingEndpoint: Used to create serving endpoints (LLM + Embedding, etc).

So I think we probably need these 2 classes.

Let me know if I am missing something